src/display/api.js

```diff
@@ -698,13 +698,21 @@ class PDFDocumentProxy {
     return this._pdfInfo.isPureXfa;
   }
 
+  /**
+   * @type {Object} An object representing a HTML tree structure to render
+   * the XFA.
+   */
+  get allXfaHtml() {
+    return this._pdfInfo.htmlForXfa;
+  }
+
   /**
    * @param {number} pageNumber - The page number to get. The first page is 1.
    * @returns {Promise<PDFPageProxy>} A promise that is resolved with
    *   a {@link PDFPageProxy} object.
    */
   getPage(pageNumber) {
-    return this._transport.getPage(pageNumber);
+    return this._transport.getPage(pageNumber, this._pdfInfo.htmlForXfa);
```

Comment:
I'd suggest just adding `this._htmlForXfa = pdfInfo.htmlForXfa;` after line https://github.com/mozilla/pdf.js/blob/3538ef017fa05787d2e0c2b6bdf2d904753da489/src/display/api.js#L2458 instead, to avoid having to pass this property around as much.